pkg/controller/controller_instance.go

```diff
@@ -801,7 +801,11 @@ func (c *controller) reconcileServiceInstance(instance *v1beta1.ServiceInstance)
 		// Only send the parameters if they have changed from what the Broker has
 		if toUpdate.Status.ExternalProperties == nil ||
 			toUpdate.Status.InProgressProperties.ParametersChecksum != toUpdate.Status.ExternalProperties.ParametersChecksum {
-			updateRequest.Parameters = parameters
+			if parameters != nil {
+				updateRequest.Parameters = parameters
+			} else {
+				updateRequest.Parameters = make(map[string]interface{})
```

Comment:
We talked about the question of how to get back to the defaults in the SIG. Would the update like this allow for defaults after the update is applied or are the properties set to nil/zero/empty?

Reply:
I don't quite understand what you are asking? Would you mind rephrasing the question to help get it through my head?

Reply:
It seems like @n3wscott is asking that we debated how to reset params to the defaults (empty in this case). If I recall correctly, that was a discussion on how to empty out fields in the Kubernetes resources, but this line is resetting parameters in an OSB API request, and is the only way to do so.

@n3wscott am I on the same track as what you're thinking?

Reply:
This continues the approach that we have taken to date with regards to update instance parameters. This may (and likely will) change in the future. For now, when a parameter is deleted, we send all the parameters that were sent before save for the deleted parameter, which is omitted. Extended that to its logical conclusion, when we delete all parameters, we should send an empty JSON object. Currently in master, we omit the Parameters field, which the Broker **must** interpret as no change to the Parameters.

Reply:
Yeah, in my past experience it was always extremely difficult to understand if the intent is to set a value to null, 0, or "" OR allow the default to show through and remove the value set. We solved this by adding to the update request a way to say which path value you would like to be deleted explicitly and that helped with the confusion. 

It could be easier in golang because you can have nil ints? in java land it was hard to know if that int was zero because you want it to be zero or deleted.

Reply:
I am fine with this as is.